tools/cloud-function-automation/cloud_function_automation.py

```diff
@@ -0,0 +1,278 @@
+        except Exception as exc:
+            raise SystemExit(exc)
+        print(f"Log Router Sink {sink.name} has been created")
+        created_sink = True
+    return created_sink
+
+
+def create_cloud_function(function_config, project_id, full_topic_name):
+    """
+    This function will be used to create and deploy cloud function.
+    :param function_config: dictionary containing cloud function configurations
+    mentioned in variables.py.
+    :param project_id: Google Cloud project id mentioned in variables.py.
+    :param full_topic_name: Pub/Sub topic associated with Cloud function.
+    :return:
+    """
+    function_zip_file_path = function_config["CLOUD_FUNCTION_ZIP_FILE_PATH"]
+    function_location = function_config["CLOUD_FUNCTION_LOCATION"]
+    function_name = function_config["CLOUD_FUNCTION_NAME"]
+    function_runtime = function_config["CLOUD_FUNCTION_RUNTIME"]
+    full_location = f"projects/{project_id}/locations/{function_location}"
+    # Create a client
+    try:
+        client = functions_v1.CloudFunctionsServiceClient()
+    except Exception as exc:
+        raise SystemExit(exc)
+
+    # Initialize request argument(s)
+    function = functions_v1.CloudFunction()
+    function.source_archive_url = function_zip_file_path
+    function.name = f"projects/{project_id}/locations/{function_location}/functions/{function_name}"
+    function.event_trigger = {"event_type": "google.pubsub.topic.publish",
+                              "resource": full_topic_name, "service": "pubsub.googleapis.com"}
+    function.runtime = function_runtime
+
+    request = functions_v1.CreateFunctionRequest(
+        location=full_location,
+        function=function
+    )
+
+    # Make the request
+    try:
+        operation = client.create_function(request=request)
+    except Exception as exc:
+        raise SystemExit(exc)
+
+    print("Waiting for operation to complete...")
+
+    response = operation.result()
+
+    # Handle the response
+    print(response)
+    print(f"Cloud function {function_name} has been created")
+
+
+def delete_cloud_pubsub_topic(project_id, topic_id):
+    """
+    This function will be used to delete Cloud Pub/Sub Topic if exist as per variables.py.
+    :param project_id: Google Cloud project id mentioned in variables.py.
+    :param topic_id: Pub/Sub topic associated with Cloud function.
+    :return:
+    """
+    try:
+        publisher = pubsub_v1.PublisherClient()
+    except Exception as exc:
+        raise SystemExit(exc)
+    topic_path = publisher.topic_path(project_id, topic_id)
+    try:
+        publisher.delete_topic(request={"topic": topic_path})
+        print(f"Topic deleted: {topic_path}")
+    except Exception as exc:
+        print(exc)
+
+
+def delete_cloud_log_sink(sink_name):
+    """
+    This function will be used to delete GCP Log Sink.
+    :param sink_name: Log Sink name.
+    :return:
+    """
+    try:
+        logging_client = logging.Client()
+    except Exception as exc:
```

Comment:
Same here (comments was: What kind of exception can happen here? Handling a generic exception and then exiting immediately does not do much. Please handle the exceptions that you would know how to handle, or at least, that you could enrich with some more meaningful error message.)